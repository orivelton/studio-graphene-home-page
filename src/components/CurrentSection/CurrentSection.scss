@import '../../assets/scss/helper.scss';

@mixin transition-width {
  -webkit-transition: width 0.5s ease-in-out;
  -moz-transition: width 0.5s ease-in-out;
  -o-transition: width 0.5s ease-in-out;
  transition: width 0.5s ease-in-out;
}

.count-section {
  position: fixed;
  display: flex;
  flex-direction: column;
  top: 297px;
  left: 0;
  z-index: 5;
}

.current-section {
  @include transition-width;
  width: 15px;
  height: 18px;
  font-size: 18px;
  display: flex;
  left: 27px;
  align-items: center;
  position: relative;

  &::before {
    content: '';
    position: absolute;
    top: 6px;
    background: $night-rider;
    display: flex;
    width: 100%;
    height: 2px;
    opacity: 1;
  }
  
  &--double {
    @include transition-width;
    width: 30px;
  }
}
