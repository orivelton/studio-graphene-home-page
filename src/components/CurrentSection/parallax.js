import { getSection } from "../../helpers";

const parallax = () => {
  const position = -getSection()[1].getBoundingClientRect().top;
  const style = `
    margin-top: ${position}px;
    transition: margin 0.8s ease-in-out
  `;
    
  getSection('.list-menu__type').forEach((item, index) => {
    if(index === 1 || index === 3) item.setAttribute('style', style);
  });
}

export { parallax };