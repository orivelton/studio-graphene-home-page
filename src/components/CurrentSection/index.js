import React, { useEffect, useState } from 'react';
import './CurrentSection.scss';

const CurrentSection = () => {
  const [section, setSection] = useState(1);

  const getSection = () => [...document.querySelectorAll('section')];
  
  const handleScroll = () => {
    const screen = document.documentElement.clientHeight / 2 - 100;
    const sections = getSection();
    sections.forEach((item, index) => {
      const { top, bottom } = item.getBoundingClientRect();
      top <= screen && bottom >= screen && setSection(++index);
    });
  }

  useEffect(() => window.addEventListener('scroll', handleScroll), []);

  return (
    <div className="count-section">
      {
        [...Array(4)].map((item, index) => {
          const isCurrentSectio = section === ++index;

          return(
            <span key={`${index}${index}`} className={isCurrentSectio ? 'current-section current-section--double' : 'current-section'}>
              { isCurrentSectio && ('0' + section).slice(-2)}
              
            </span>
          )
        })
      }
    </div>
  );
};

export default CurrentSection;
