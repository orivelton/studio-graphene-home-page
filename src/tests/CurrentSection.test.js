import React from 'react';
import renderer from 'react-test-renderer';
import CurrentSection from '../components/CurrentSection';

test('renders correctly CurrentSection', () => {
  const tree = renderer.create(<CurrentSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
