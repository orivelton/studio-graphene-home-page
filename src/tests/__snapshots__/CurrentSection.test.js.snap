// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly CurrentSection 1`] = `
<div
  className="count-section"
>
  <span
    className="current-section current-section--double"
  >
    01
  </span>
  <span
    className="current-section"
  />
  <span
    className="current-section"
  />
</div>
`;
